pip/req/req_uninstall.py

```diff
@@ -69,6 +70,20 @@ def add_pth(self, pth_file, entry):
         else:
             self._refuse.add(pth_file)
 
+    def add_dist_record(self):
+        """Add paths for all the files in self.dist's RECORD.
+        For each .py file in RECORD, add the .pyc in the same directory.
+        """
+        r = csv.reader(FakeFile(self.dist.get_metadata_lines('RECORD')))
+        for row in r:
+            path = os.path.join(self.dist.location, row[0])
+            self.add(path)
+            if path.endswith('.py'):
+                dn, fn = os.path.split(path)
+                base = fn[:-3]
+                path = os.path.join(dn, base + '.pyc')
+                self.add(path)
```

Comment:
I know this is just a shuffle refactor, but is this bit needed anymore?  

compiled files now end up in RECORD during install.

if it is needed for some special case, better to move it over into `UninstallPathSet.add` where it can be done in a general way.  adding pyc's is pointless for example on py3.